bentoml/_internal/bento/build_config.py

```diff
@@ -54,98 +63,207 @@ def _convert_python_version(py_version: t.Optional[str]) -> t.Optional[str]:
     return f"{major}.{minor}"
 
 
+def _convert_cuda_version(cuda_version: t.Optional[str]) -> t.Optional[str]:
+    if cuda_version is None or cuda_version == "":
+        return None
+
+    if isinstance(cuda_version, int):
+        cuda_version = str(cuda_version)
+
+    if cuda_version == "default":
+        cuda_version = DEFAULT_CUDA_VERSION
+
+    match = re.match(r"^(\d+)", cuda_version)
+    if match is not None:
+        cuda_version = SUPPORTED_CUDA_VERSION[cuda_version]
+
+    return cuda_version
+
+
+def _convert_user_envars(
+    envars: t.Optional[t.Union[t.List[str], t.Dict[str, t.Any]]]
+) -> t.Optional[t.Dict[str, t.Any]]:
+    if envars is None:
+        return None
+
+    if isinstance(envars, list):
+        return {k: v for e in envars for k, v in e.split("=")}
+
+    return envars
+
+
+def _envars_validator(
+    _: t.Any,
+    attribute: "Attribute[t.Union[t.List[str], t.Dict[str, t.Any]]]",
+    value: t.Union[t.List[str], t.Dict[str, t.Any]],
+) -> None:
+    if isinstance(value, list):
+        for envar in value:
+            try:
+                k, _ = envar.split("=")
+                if not k.isupper():
+                    raise InvalidArgument(f"{k} should be in uppercase.")
+            except ValueError as ve:
+                raise InvalidArgument(
+                    f"{envar} doesn't follow format ENVAR=value"
+                ) from ve
+    elif isinstance(value, dict):
+        if not all(k.isupper() for k in value):
+            raise InvalidArgument("All envars should be in UPPERCASE.")
+    else:
+        raise InvalidArgument(
+            f"`env` must be either a list or a dict, got type {attribute.type} instead."
+        )
+
+
+class OptionsMeta:
```

Comment:
Why `Meta`? Also, should this be a subclass of `ABC`?

Reply:
actually it is just an interface,